intents/main.py

```diff
@@ -0,0 +1,186 @@
+"""
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+"""
+This module contains the function that gets called by the UI, and it
+serves a purpose of a master function to all the intets.
+It returns the result of the intent and also the list of suggestions.
+"""
+
+import json, pandas, enum, topk
+from flask import escape
+from show import show
+from util import enums
+
+def hello_http(request):
+    """HTTP Cloud Function.
+    Args:
+        request (flask.Request): The request object.
+        <http://flask.pocoo.org/docs/1.0/api/#flask.Request>
+    Returns:
+        The response text, or any set of values that can be turned into a
+        Response object using `make_response`
+        <http://flask.pocoo.org/docs/1.0/api/#flask.Flask.make_response>.
+    """
+    request_json = request.get_json(silent=True)
+    request_args = request.args
+
+    # extracting the intent parameters from the json
+    intent = request_json['intent']
```

Comment:
Can we maybe define a dictionary where key becomes key of each field in request_json and the value is its value? But can leave as is if this is more efficient.

Reply:
But I think each time we would access dict_name['key_name'] and that is okay too, but this seems more efficient to me. But can change, as it won't be modified and all changes will be in the intents files.